tests/release-notes-test.ts

```diff
@@ -0,0 +1,69 @@
+jest.mock("danger", () => jest.fn());
+import danger from "danger";
+const dm = danger as any;
+
+import checkReleaseNotes from "../org/pr/release-notes";
+
+// The mocked data and return values for calls the rule makes.
+beforeEach(() => {
+    dm.warn = jest.fn().mockReturnValue(true);
+
+    dm.danger = {
+        git: {
+            modified_files: ["RELEASE-NOTES.txt"],
+            created_files: [],
+            deleted_files: []
+        },
+        github: {
+            pr: {
+                base: {
+                    ref: "release/1.0",
+                }
```

Comment:
I think it could be useful to consider more test cases for some other PR branch configurations, including:

 - `head: { ref: "release/1.0" }, base: { ref: "develop" }` : a PR from release to develop, which should be _allowed_ to have RELEASE-NOTES.txt changes (especially after we update them once Editorial get back to us
 - `base: { ref: "release/merge-1.0-to-develop" }` : type of PRs we use for conflict resolution. Do we want the rule to forbid `RELEASE-NOTES.txt` changes for those? Though not sure what most people use as naming convention for those, we might not be all consistent, but it's worth including a Unit Test for imho to at least not forget that use case and test our expected behavior against it
 - `base: { ref: "hotfix/1.0.1" }`: shouldn't we enable the check (and forbid release notes changes) for PRs targeting a hotfix like we do for a release?

Reply:
> head: { ref: "release/1.0" }, base: { ref: "develop" } : a PR from release to develop, which should be allowed to have RELEASE-NOTES.txt changes (especially after we update them once Editorial get back to us)

There's already a test with `develop` as the base branch and the rule doesn't mind the `head` branch at all, so isn't it enough? 🤔

> base: { ref: "release/merge-1.0-to-develop" } : type of PRs we use for conflict resolution. Do we want the rule to forbid RELEASE-NOTES.txt changes for those?

The rule doesn't mind the `head` branch, so I'm not sure this would be useful. 

> shouldn't we enable the check (and forbid release notes changes) for PRs targeting a hotfix like we do for a release?

Yup! I believe nobody expects the release-notes to be translated for a hotfix... but I agree that adding it doesn't hurt. I've updated the PR.

Reply:
> the rule doesn't mind the head branch at all, so isn't it enough? 🤔

Well strictly speaking, saying that means you rely on the current implementation details to write your test (as opposed to thinking TDD 😅 ). Which means that if for whatever reason we refactor the rule a bit in the future, thinking we'd improve its performance without changing the behavior, we could accidentally check the wrong branch after our refactor, or use e.g. an hypothetical new method like `github.pr.branches.contains('release')` new property, instead of specifically `head` or `base`, and break the behavior without the test catching it 😉

Reply:
> Well strictly speaking, saying that means you rely on the current implementation details to write your test 

Well, tbh, I think it's the opposite: I'm not looking at the implementation details at all ;-) Let's step back a bit: this rule is about checking that a certain type of changes  don't get merged into a release branch. From where they come, it doesn't matter at all (if you don't think about the implementation), because there are no special cases where we want this behaviour to change depending on the head branch.
But I agree that if you look at the implementation details, you may be worried that a branch which has `release` in the name can become a problem in the future if we make a mistake with the code :-)